src/components/widgets-common/chart-threshold/chart-threshold.component.coffee

```diff
@@ -67,14 +67,21 @@ module.component('chartThreshold', {
       , 100)
       return
 
+    handleInvalidAlertAmount = ->
+      toastr.error("Please choose a number one or greater.", 'Error')
+      ctrl.cancelCreateKpi()
```

Comment:
Is this new functionality? Do we want to `cancelCreateKpi` if I enter a wrong number?

Reply:
Yes we do, currently when we enter a wrong number, the update box will keep on spinning.

Reply:
@iseessel I had a quick look, the problem is the `ctrl.saveKpi` method sets `ctrl.loading = true`, but then returns without setting the `ctrl.loading` back to `false` unless `ImpacKpisSvc.validateKpiTargets(params.targets)`.

The quick solution to use a toastr to notify the user is fine I guess, though ngMessages txt would probs be better. But instead of calling `ctrl.cancelCreateKpi()` in your `handleInvalidAlertAmount` method, can we just try setting `ctrl.loading` to `false`? I think it would be a better ux.

Reply:
agreed